pkg/metadata/provider_aws.go

```diff
@@ -61,6 +80,19 @@ func (p *ProviderAWS) Extract() ([]byte, error) {
 	return userData, nil
 }
 
+// lookup a value (lookupName) in aws metaservice and store in given fileName
+func awsMetaGet(lookupName string, fileName string, fileMode os.FileMode) {
+	lookupValue, err := awsGet(metaDataURL + lookupName)
+	if err == nil {
+		err = ioutil.WriteFile(path.Join(ConfigPath, fileName), lookupValue, fileMode)
```

Comment:
I thought about that, but didn't know what the current thinking was around this. Some of the data is AWS specific (availability zone, instance id, instance type) some isn't (public ip, local ip, local hostname). 

If you want I can change it to `/var/config/aws/<blah>` for the aws stuff, and leave the other fields the way they are.